internal/sqldb/sqlquery/statvar_summary.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlquery
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+	"github.com/datacommonsorg/mixer/internal/util"
+)
+
+// GetStatVarSummary returns summaries of the specified statvars.
+func GetStatVarSummary(sqlClient *sql.DB, statvars []string) (map[string]*pb.StatVarSummary, error) {
+	defer util.TimeTrack(time.Now(), "SQL: GetStatVarSummary")
+
+	result := map[string]*pb.StatVarSummary{}
+
+	if len(statvars) == 0 {
+		return result, nil
+	}
+
+	query := getStatVarSummarySqlQuery(statvars)
+	rows, err := sqlClient.Query(query, util.ConvertArgs(statvars)...)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var variable, entityType, sampleEntityIds string
+		var entityCount int32
+		var minValue, maxValue float64
+
+		err = rows.Scan(&variable, &entityType, &entityCount, &minValue, &maxValue, &sampleEntityIds)
+		if err != nil {
+			return nil, err
+		}
+		if _, ok := result[variable]; !ok {
+			result[variable] = &pb.StatVarSummary{
+				PlaceTypeSummary: map[string]*pb.StatVarSummary_PlaceTypeSummary{},
+			}
+		}
+
+		summary := result[variable]
+		placeTypeSummary := pb.StatVarSummary_PlaceTypeSummary{}
+		summary.PlaceTypeSummary[entityType] = &placeTypeSummary
+		placeTypeSummary.PlaceCount = entityCount
+		placeTypeSummary.MinValue = &minValue
+		placeTypeSummary.MaxValue = &maxValue
+
+		entityIds := strings.Split(sampleEntityIds, ",")
+		for _, entityId := range entityIds {
+			place := pb.StatVarSummary_Place{
+				Dcid: entityId,
+				Name: entityId,
+			}
+			placeTypeSummary.TopPlaces = append(placeTypeSummary.TopPlaces, &place)
```

Comment:
nit: can gather all top places as a list before line 59 and assign it in the above block at once.